### PR TITLE
A logic patch and optimization 

### PR DESCRIPTION
**Pull request by BDXGD**
**What type of PR is this?**

/kind cleanup

**What this PR does / why we need it**:

The same as #978.

### CODE REVIEW

**Comment by RainbowMango** [comment]
Sorry for letting this sit so long.

@XiShanYongYe-Chang Please help look at it.

File: pkg/util/binding.go [hunk 1]
```diff
@@ -76,16 +76,33 @@ func DivideReplicasByTargetCluster(clusters []workv1alpha2.TargetCluster, sum in
...
+}
+
+// divideRemainingReplicas divide remaining Replicas to clusters
+func divideRemainingReplicas(remainingReplicas int, res []workv1alpha2.TargetCluster) {
+	if remainingReplicas <= 0 {
+		return
+	}
+
+	clusterSize := len(res)
+	if remainingReplicas < clusterSize {
```
Comment (XiShanYongYe-Chang) [inline pkg/util/binding.go:92]: When the value of `remainingReplicas` is several times the value of `clusterSize`, some calculations can be omitted.

Maybe we can remove judgment. If match the case `remainingReplicas < clusterSize`, the `avg` value is 0.

File: pkg/util/binding.go [hunk 2]
```diff
@@ -76,16 +76,33 @@ func DivideReplicasByTargetCluster(clusters []workv1alpha2.TargetCluster, sum in
...
+		for i := 0; i < remainingReplicas; i++ {
 			res[i].Replicas++
-			remainReplicas--
+		}
+	} else {
+		avg, residue := remainingReplicas/clusterSize, remainingReplicas%clusterSize
```
Comment (XiShanYongYe-Chang) [inline pkg/util/binding.go:97]: We can convert the int type to int32 here, so as not to convert one by one during the loop.

**Comment by XiShanYongYe-Chang** [comment]
kindly ping @BDXGD

**Comment by RainbowMango** [comment]
/close
Feel free to re-open if you think this is still needed.

**Pull request closed without merging**